src/PlaywrightSharp.Abstractions/IPage.cs

```diff
@@ -144,8 +151,16 @@ public interface IPage
         /// ]]>
         /// </code>
         /// </example>
-        event EventHandler<ConsoleEventArgs> Console;
-
+        event EventHandler<ConsoleEventArgs> Console;
+
+        /// <summary>
+        /// Waits for event to fire and passes its value into the predicate function.
+        /// </summary>
+        /// <param name="e">Event to wait for</param>
+        /// <param name="options">Extra options</param>
+        /// <returns>A <see cref="Task"/> that completes when the predicate returns truthy value. Yielding the information of the event.</returns>
+        Task<T> WaitForEvent<T>(PageEvent e, WaitForEventOptions options = null);

```

Comment:
@Meir017 do you have a better/generic idea for this?
It sucks that we can't make extension methods over events.

Reply:
Maybe instead we could expose a generic event property?
```cs
event EventHandler<PageEventArgs> Event;
```
where `PageEventArgs` will be something like
```cs
public class PageEventArgs
{
    public string PageEventName { get; } // enum
    public object Data { get; } // can by anything...
}
```